### PR TITLE
=act add benchmark of ByteString#take(...)

### PR DESCRIPTION
**Pull request by monkey-mas**
Added a benchmark code to measure use-case performance.

ref. https://github.com/akka/akka/pull/21438

### CODE REVIEW

**Review by johanandren — APPROVED** [review]
Thanks!

**Comment by ktoso** [comment]
OK TO TEST

**Comment by ktoso** [comment]
Thanks a lot :)

File: akka-bench-jmh/src/main/scala/akka/util/ByteString_take_Benchmark.scala [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+   [info] ByteString_take_Benchmark.bss_best                  thrpt   40   197237882.251 ±  2714956.732  ops/s
+   [info] ByteString_take_Benchmark.bss_greater_or_eq_to_len  thrpt   40   866558812.838 ± 15343155.818  ops/s
+   [info] ByteString_take_Benchmark.bss_negative              thrpt   40  1114723770.487 ± 30945339.512  ops/s
+   [info] ByteString_take_Benchmark.bss_worst                 thrpt   40      233870.585 ±     7326.227  ops/s
+
+   */
+
+  @Benchmark
+  def bss_negative(): Unit = {
+    @volatile var bs: ByteString = null
```
Comment (johanandren) [inline akka-bench-jmh/src/main/scala/akka/util/ByteString_take_Benchmark.scala]: Isn't this the typical case where you should use `blackhole.consume`?

**Review by johanandren — APPROVED** [review]

File: akka-bench-jmh/src/main/scala/akka/util/ByteString_take_Benchmark.scala (hunk 1 above)
Comment (ktoso) [inline akka-bench-jmh/src/main/scala/akka/util/ByteString_take_Benchmark.scala]: Yes, correct, in fact just returning would be better.

File: akka-bench-jmh/src/main/scala/akka/util/ByteString_take_Benchmark.scala [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+  @Benchmark
+  def bss_best(): ByteString = {
+    bss.take(n_best)
+  }
+
+  @Benchmark
+  def bss_worst(): ByteString = {
+    bss.take(n_worst)
+  }
```
Comment (ktoso) [inline akka-bench-jmh/src/main/scala/akka/util/ByteString_take_Benchmark.scala]: @johanandren I amended this PR to return the values here, thanks for spotting - let's merge after validation

**Review by ktoso — APPROVED** [review]

**Comment by johanandren** [comment]
Wat, didn't know you could do that!

**Comment by ktoso** [comment]
Yep, since this feature came in: https://github.com/blog/2247-improving-collaboration-with-forks

**Comment by ktoso** [comment]
Failure was FramingSpec timeout, I believe we track this already: https://github.com/akka/akka/issues/21673

**Comment by ktoso** [comment]
Benchmark always nice to have, thanks!

**Pull request merged**